mono/utils/mono-errno.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * \file
+ * Access the native error code
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+
+#ifndef __MONO_ERRNO_H__
+#define __MONO_ERRNO_H__
+
+// Enough indirection to do something else here, or log
+
+#define SET_ERRNO(errno_val) do { \
```

Comment:
Inline function? So I can set breakpoint? Expression? Getter not just setter? And no trailing semicolon. But good idea.

Reply:
And it should be rare enough, that a non-inline function should suffice. Errno is a thread local, and accesses might involve a function call, or two. On Windows, probably two, try stepping through it.

Reply:
i.e. one of:
`#define SET_ERRNO(x) (errno = (x))`

or:
```
void mono_set_errno(inline x); // .h  inline or not 
void mono_set_errno(int x) { errno = x; ] // .c
```

A macro that is well behaved (no multi-evaluation of parameters) imho could just be named like a function, though I not sure mono allows that.

I'd just as soon put `MONO_` or `mono_` prefix on, even if there no linkage (macro).
i.e. what if later it becomes a function (granted `#define SET_ERRNO mono_set_errno`).